### PR TITLE
Unsupervised ML guide

### PR DESCRIPTION
**Pull request by moonlanderr**
\<insert pull request description here\>

-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [ ] All `import`s are in the first cell? First block of imports are standard libraries, second block are 3rd party libraries, third block are all `arcgis` imports? Note that in some cases, for samples, it is a good idea to keep the imports next to where they are used, particularly for uncommonly used features that we want to highlight.
- [ ] All `GIS` object instantiations are one of the following?
    - `gis = GIS()`
    - `gis = GIS('https://www.arcgis.com', 'arcgis_python', 'P@ssword123')`
    - `gis = GIS(profile="your_online_profile")`
    - `gis = GIS('https://pythonapi.playground.esri.com/portal', 'arcgis_python', 'amazing_arcgis_123')`
    - `gis = GIS(profile="your_enterprise_portal")`
- [ ] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`?
- [ ] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the api\_data\_owner user?
- [ ] Code refactored & split out across multiple cells, useful comments?
- [ ] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [ ] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [ ] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.)
- [ ] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ DavidJVitale so he can add it to the list for the next deploy

### CODE REVIEW

**Comment by priyankatuteja** [comment]
@moonlanderr Could you review this one?

File: guide/14-deep-learning/unsupervised-ml-model-guide.ipynb [hunk 1]
```diff
@@ -0,0 +1,609 @@
+{
```
Comment (AngadKundra) [inline guide/14-deep-learning/unsupervised-ml-model-guide.ipynb:1]: <p>Data can be feature layer, spatially enabled dataframe with/without rasters or just a simple dataframe or rasters.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/871/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (AngadKundra) [inline guide/14-deep-learning/unsupervised-ml-model-guide.ipynb:1]: <p>[("Field_1", "Field_2",......."Field_n", MinMaxScaler())]</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/871/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (AngadKundra) [inline guide/14-deep-learning/unsupervised-ml-model-guide.ipynb:1]: <p>[('raster.tif',MinMaxScaler())]&nbsp;</p><p>or [(raster.name, MinMaxScaler())] -&gt; raster is the Raster object</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/871/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (AngadKundra) [inline guide/14-deep-learning/unsupervised-ml-model-guide.ipynb:1]: <p>input_features can be simple dataframe too.</p><p>explanatory_rasters - List of the Raster objects, where 2-sized tuples has to be passed for categorical variables as discussed above.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/871/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

**Review by AngadKundra — APPROVED** [review]
Looks good to me.

File: guide/14-deep-learning/unsupervised-ml-model-guide.ipynb [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
```
Comment (priyankatuteja) [inline guide/14-deep-learning/unsupervised-ml-model-guide.ipynb:1]: <p>it should be as a cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/871/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

**Review by priyankatuteja — APPROVED** [review]

**Comment by mohi9282** [comment]
@moonlanderr Could you please create a reference section towards the end of the notebook and list references there? This guide talks about various models so a reference section would help.

**Comment by moonlanderr** [comment]
> @mohi9282 reference section added. Please check.

**Review by mohi9282 — APPROVED** [review]
lgtm

cc: @AtmaMani

**Review by AtmaMani — APPROVED** [review]

**Pull request merged**